src/Microsoft.Diagnostics.Monitoring.WebApi/ProcessInfoImpl.cs

```diff
@@ -144,6 +131,23 @@ public static async Task<IProcessInfo> FromEndpointInfoAsync(IEndpointInfo endpo
                 processName);
         }
 
+        public static bool IsWindowsProcess(IEndpointInfo endpointInfo)
+        {
+            if (string.IsNullOrEmpty(endpointInfo.OperatingSystem))
+            {
+                // If operating system is null, the process is likely .NET Core 3.1 (which doesn't have the GetProcessInfo command).
```

Comment:
I was also thinking the same thing, but decided to keep the existing logic since I don't have the full context of whether we need to use this check or not.

Reply:
At this time, it's safe to assume `dotnet-monitor` does run on the same platform as the target process. The primary scenarios all require that they are running on the same machine.

The diagnostic communication channel is either a named pipe or a unix domain socket. UDS are inter-process communication within the same host machine. A Windows named pipe does technically allow for communication over a network, but that is not something that we've explicitly enabled (running `dotnet-monitor` on a machine different that the target process).